File: backend/app/api/schemas.py
"""Pydantic schemas for minimal v1 endpoints."""

from pydantic import BaseModel, Field

from app.agent.video_styles import VideoStyle


class VideoCreateRequest(BaseModel):
    prompt: str = Field(..., min_length=1)
    video_style: VideoStyle = Field(
        default=VideoStyle.GENERAL,
        description="The video production style to apply during prompt enhancement.",
    )


class VideoCreateResponse(BaseModel):
    job_id: str
    status: str
    output_path: str | None = None
    job_project_path: str | None = None
    error: str | None = None

File: backend-ts/src/schemas.ts
import { z } from "zod";
import type {
  UploadedFile,
  VideoCreateRequest,
  VideoCreateResponse,
} from "@shared/video-contract";
import { VideoStyle } from "./agent/video-styles.js";

const videoStyleValues = [VideoStyle.GENERAL, VideoStyle.TRAILER] as const;

export const videoCreateRequestSchema: z.ZodType<VideoCreateRequest> = z.object({
  prompt: z.string().min(1),
  video_style: z.enum(videoStyleValues).default(VideoStyle.GENERAL),
});

export const videoCreateResponseSchema: z.ZodType<VideoCreateResponse> = z.object({
  job_id: z.string(),
  status: z.enum(["complete", "failed"]),
  output_path: z.string().nullable().optional(),
  job_project_path: z.string().nullable().optional(),
  error: z.string().nullable().optional(),
});

export const uploadedFileInfoSchema: z.ZodType<UploadedFile> = z.object({
  name: z.string(),
  size: z.number(),
  type: z.string(),
  description: z.string(),
  uploaded_at: z.string(),
  has_thumbnail: z.boolean(),
});

export type { VideoCreateRequest, VideoCreateResponse, UploadedFile };
